eth/api.go

```diff
@@ -438,10 +441,15 @@ func (s *PrivateAccountAPI) NewAccount(password string) (common.Address, error)
 	return common.Address{}, err
 }
 
-// UnlockAccount will unlock the account associated with the given address with the given password for duration seconds.
-// It returns an indication if the action was successful.
-func (s *PrivateAccountAPI) UnlockAccount(addr common.Address, password string, duration int) bool {
-	if err := s.am.TimedUnlock(addr, password, time.Duration(duration)*time.Second); err != nil {
+// UnlockAccount will unlock the account associated with the given address with
+// the given password for duration seconds. If duration is nil it will use a
+// default of 300 seconds. It returns an indication if the account was unlocked.
+func (s *PrivateAccountAPI) UnlockAccount(addr common.Address, password string, duration *rpc.HexNumber) bool {
+	if duration == nil {
+		duration = rpc.NewHexNumber(300)
+	}
```

Comment:
As briefly discussed. This is a default when the user don't supply a duration. If the users chooses to unlock the account for ever he can supply 0 which will unlock the account forever. Have updated the docs to explain this behaviour.